Blocks/HTTP/test.cc

```diff
@@ -926,3 +926,26 @@ TEST(HTTPAPI, ResponseSmokeTest) {
     static_cast<void>(response);
   }
 }
+
+TEST(HTTPAPI, PayloadTooLarge) {
+  const auto scope = HTTP(FLAGS_net_api_test_port)
+                         .Register("/enough_is_enough",
+                                   [](Request r) {
+                                     ASSERT_FALSE(r.body.empty());
+                                     EXPECT_EQ("POST", r.method);
+                                     r("Fits.\n");
+                                   });
+
+  {
+    const auto response = HTTP(POST(Printf("http://localhost:%d/enough_is_enough", FLAGS_net_api_test_port),
+                                    std::string(16 * 1024 * 1024, '.')));
```

Comment:
@mzhurovich, :+1: , pushed this, and a broadened `catch` to see if it fixes the tests on Mac.